CHANGELOG.md

```diff
@@ -1,6 +1,7 @@
 ## vNext (TBD)
 
 ### Enhancements
+* Removed backing fields of Realm Objects' properties which should provide minor improvements to memory used by Realm Objects (Issue [#2647](https://github.com/realm/realm-dotnet/issues/2994))
```

Comment:
I think we should specify this is impacting only generated classes, not `RealmObject` inheritors.